x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/plan/logical/JoinSerializationTests.java

```diff
@@ -27,7 +28,7 @@ protected Join createTestInstance() {
     }
 
     private static JoinConfig randomJoinConfig() {
-        JoinType type = randomFrom(JoinType.values());
+        JoinType type = randomFrom(JoinTypes.LEFT, JoinTypes.RIGHT, JoinTypes.INNER);
```

Comment:
Why do we restrict the join types in this serialization test?

Reply:
Because natural and using don't survive analysis and thus don't/shouldn't get serialized.